nncf/sparsity/magnitude/algo.py

```diff
@@ -41,15 +41,13 @@ def __init__(self, target_model: NNCFNetwork,
                  sparsified_module_info: List[SparseModuleInfo],
                  params, weight_importance: str):
         super().__init__(target_model, sparsified_module_info)
-        self.sparsity_level = self.threshold = 0
         self.weight_importance = WEIGHT_IMPORTANCE_FUNCTIONS.get(weight_importance)
         scheduler_cls = SPARSITY_SCHEDULERS.get(params.get("schedule", "polynomial"))
         self._scheduler = scheduler_cls(self, params)
 
     def statistics(self):
         stats = super().statistics()
-        stats['sparsity_threshold'] = self.threshold
-        stats['sparsity_level'] = self.sparsity_level
```

Comment:
why did you remove sparsity_level?

Reply:
This `sparsity_level` report only served to increase confusion. This is not the actual sparsity level, but rather the level that the sparsity scheduler will set at the next call of "set_sparsity_level" (don't remember the exact name). The actual sparsity rate statistic is already reported as "sparsity_rate_for_sparsified_modules". As a result, during evaluation of a completely sparsified checkpoint and calling statistics the user would get seemingly contradictive values of "sparsity_level" (which would be set to some initial scheduled value as specified in the config file) and "sparsity_rate_for_sparsified_modules" (which would be close to the final sparsity value of the config file).

Reply:
yep, there's also `sparsity_rate_for_model` to represent sparsity level for the whole model. then it should be ok